File: src/main/resources/static/js/app/emailCreate.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */

app.controller("emailCreate", function ($scope, $http, $mdDialog, $filter) {

    $scope.mailCreate = {

        rqstId: "",
        epfNo: "",
        userName: "",
        fullName: "",
        designation: "",
        emailAddress: "",
        branchorDept: "",
        profile: "",
        password: "",
        mobileNo: "",
        click: "",
        status: "",
        currentTill: "",
        click2: "",
        newBranchOrDept: "",
        newProfile: "",
        newDesignation: "",
        newTill: "",
        reason: "",
        startDate: "",
        endDate: "",
        rqstType: ""


    };

    $scope.mailCreate.toDay = $filter('date')(new Date(), 'MM/dd/yyyy');

    $scope.cofrmSubmition = function (value) {

        if (value == true) {

            $("#cofrmTypModal").modal("hide");
            $("#createEmail").modal("hide");

            var method = "POST";
            var url = 'emlctrt/confrm_sub';
            $scope.createEmailBusy = $http({
                method: method,
                url: url,
                data: angular.toJson($scope.mailCreate),
                headers: {
                    'Content-Type': 'application/json'
                },
                transformResponse: [
                    function (data) {

                        var msg = $scope.getEmailToCreate();
                        if (data == "Request Fail") {
                            sweetAlert("Alert..!", data, "error");
                        } else {
                            sweetAlert("Alert..!", data, "success");
                        }

                    }
                ]
            });
        } else {

            $("#cofrmTypModal").modal("hide");
            $("#createEmail").modal("hide");
        }
    };

    $scope.getEmailToCreate = function () {


        $scope.assignData = [];
        var method = "GET";
        var url = 'emlctrt/get_users';
        $scope.createEmailBusy = $http({
            method: method,
            url: url,

            headers: {
                'Content-Type': 'application/json'
            },
            transformResponse: [
                function (data) {
                    $scope.assignData = JSON.parse(data);

                }
            ]

        });
    };

    $scope.getDataByRqstId = function (epf) {

        $scope.mailCreate.epfNo = epf;
        var method = "POST";
        var url = 'emlctrt/rqst_bydate';
        $scope.createEmailBusy = $http({
            method: method,
            url: url,
            data: angular.toJson($scope.mailCreate),
            headers: {
                'Content-Type': 'application/json'
            },
            transformResponse: [
                function (data) {
                    $scope.mailCreate = JSON.parse(data);
                    $scope.mailCreate.toDay = $filter('date')(new Date(), 'MM/dd/yyyy');
                    console.log($scope.mailCreate);
                }
            ]
        });

    };

    $scope.showSaveDislog = function () {

        $("#cofrmTypModal").modal("show");
    }

});

File: src/main/java/com/mbsl/controller/EmailCreate.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mbsl.controller;

import com.mbsl.model.User;
import com.mbsl.service.EmailCreateService;
import com.mbsl.service.UserService;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;

/**
 *
 * @author MBSL2523
 */
@RestController
public class EmailCreate implements IEmailCreate {

    @Autowired
    private EmailCreateService emailCreateService;

    @Override
    public ResponseEntity<List<User>> getEmailToCreate() {

        return new ResponseEntity<>(emailCreateService.getEmailToCreate(), HttpStatus.OK);
    }

    @Override
    public ResponseEntity<User> getDataByRqstId(User mailCreate) {

        return new ResponseEntity<>(emailCreateService.getDataByRqstId(mailCreate), HttpStatus.OK);
    }

    @Override
    public ResponseEntity<String> cofrmSubmition(User mailCreate) {

        return new ResponseEntity<>(emailCreateService.cofrmSubmition(mailCreate), HttpStatus.OK);
    }

}
